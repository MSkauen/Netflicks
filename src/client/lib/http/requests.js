import {fetchJson} from "./http";

const requests = {
    getMovie: async () => await fetchJson(""),
    getAllMovies: async () => await fetchJson("http://api.tvmaze.com/shows"),
    getActionMovies: async () => await fetchJson(""),
    getComedyMovies: async () => await fetchJson(""),
    getCrimeMovies: async () => await fetchJson(""),
    getDramaMovies: async () => await fetchJson(""),
    getHorrorMovies: async () => await fetchJson(""),
    getRomanceMovies: async () => await fetchJson(""),
    getScienceFictionMovies: async () => await fetchJson(""),
    getThrillerMovies: async () => await fetchJson(""),
};

export default requests;