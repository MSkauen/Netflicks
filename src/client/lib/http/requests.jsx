import {fetchJson} from "./http";

const requests = {
    getMovie: async () => await fetchJson(""),
    getMovieBackground: async (id) => await fetchJson(`https://api.tvmaze.com/shows/${id}/images`),
    getRandomMovie: async () => await fetchJson("http://api.tvmaze.com/shows").then(
        res => res[Math.floor(Math.random() * res.length - 1)]
    ),
    getAllMovies: async () => await fetchJson("http://api.tvmaze.com/shows"),
    getActionMovies: async () => await fetchJson("http://api.tvmaze.com/shows"),
    getComedyMovies: async () => await fetchJson(""),
    getCrimeMovies: async () => await fetchJson(""),
    getDramaMovies: async () => await fetchJson(""),
    getHorrorMovies: async () => await fetchJson(""),
    getRomanceMovies: async () => await fetchJson(""),
    getScienceFictionMovies: async () => await fetchJson(""),
    getThrillerMovies: async () => await fetchJson(""),
};

module.exports = {
    requests: requests
}