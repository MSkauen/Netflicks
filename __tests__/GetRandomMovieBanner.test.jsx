import { requests } from "../src/client/lib/http/requests";
import React from "react";
import ReactDOM from "react-dom";
import { describe, expect, it } from "@jest/globals";
import { act } from "react-dom/test-utils";
import GetRandomMovieBanner from "../src/client/components/MovieBanner";
import {fetchJson} from "../src/client/lib/http/http";

const movieApi = {
    getRandomMovie: async () => await fetchJson("http://api.tvmaze.com/shows").then(
        res => res[Math.floor(Math.random() * res.length - 1)]
    ),
};

describe("get random movie banner", () => {
    it("show movie banner", async () => {
        const container = document.createElement("div");
        document.body.appendChild(container);
        await act(async () => {
            ReactDOM.render(<GetRandomMovieBanner movieApi={movieApi}/>, container);
        });

        expect(container.innerHTML).toMatchSnapshot();
    });
})
