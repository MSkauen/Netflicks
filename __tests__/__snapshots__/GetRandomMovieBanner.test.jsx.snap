// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get random movie banner show movie banner 1`] = `""`;
